tools/ci/init

```diff
@@ -101,21 +104,224 @@ def update_sdk(output_path: str, target_version: str):
     write_file(package_builder_path, package_builder_contents)
 
 
+def write_build_sh(output_path: str, framework_name: str):
+    contents = """
+#!/usr/bin/env bash
+set -e
+
+FRAMEWORK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+BUILD_DIR=$FRAMEWORK_DIR/build/distributions
+
+FRAMEWORK_NAME={{framework_name}}
+
+${FRAMEWORK_DIR}/gradlew -p ${FRAMEWORK_DIR} check distZip
+
+BOOTSTRAP_DIR=disable \\
+EXECUTOR_DIR=disable \\
+TEMPLATE_DOCUMENTATION_PATH="http://YOURNAMEHERE.COM/DOCS" \\
+TEMPLATE_ISSUES_PATH="http://YOURNAMEHERE.COM/SUPPORT" \\
+    $FRAMEWORK_DIR/tools/build_package.sh \\
+        ${FRAMEWORK_NAME} \\
+        $FRAMEWORK_DIR \\
+        -a "$BUILD_DIR/${FRAMEWORK_NAME}-scheduler.zip" \\
+        $@
+"""
+    write_file(os.path.join(output_path, "build.sh"),
+               contents.replace("{{framework_name}}", framework_name))
+
+
+def write_build_gradle(output_path: str, framework_name: str):
+    contents = """
+group 'com.mesosphere.sdk'
+version '1.1-SNAPSHOT'
+
+apply plugin: 'java'
+apply plugin: 'application'
+
+repositories {
+    jcenter()
+    mavenCentral()
+    maven {
+        url "http://downloads.mesosphere.com/maven/"
+    }
+    maven {
+        url "http://downloads.mesosphere.com/maven-snapshot/"
+    }
+}
+
+ext {
+    junitVer = "4.12"
+    systemRulesVer = "1.16.0"
+    mockitoVer = "1.9.5"
+    dcosSDKVer = "0.40.2"
+}
+
+dependencies {
+    compile "mesosphere:scheduler:${dcosSDKVer}"
+    compile "mesosphere:executor:${dcosSDKVer}"
+    testCompile "mesosphere:testing:${dcosSDKVer}"
+}
+
+
+distributions {
+    main {
+        baseName = '{{framework_name}}-scheduler'
+        version = ''
+    }
+}
+
+mainClassName = 'com.mesosphere.sdk.{{framework_name}}.scheduler.Main'
+"""
+
+    write_file(os.path.join(output_path, "build.gradle"),
+               contents.replace("{{framework_name}}", framework_name))
+
+
+def write_gradle_settings(output_path: str, framework_name: str):
+    contents = "rootProject.name = '{{framework_name}}'"
+
+    write_file(os.path.join(output_path, "settings.gradle"),
+               contents.replace("{{framework_name}}", framework_name))
+
+
+def move_java_files(output_path: str, framework_name: str):
+
+    folders = [
+             "src/test/java/com/mesosphere/sdk/template/",
+             "src/main/java/com/mesosphere/sdk/template/"]
+
+    for f in folders:
+        shutil.move(os.path.join(output_path, f),
+                    os.path.join(output_path, f[:-len("/template/")], framework_name))
+
+
+def update_files(output_path: str, framework_name: str):
+
+    template_pattern = "([^-]?)template([^_]?)"
```

Comment:
It could be that the `?` in the regex is not playing along with what we want to do. (i.e. replacing `-template=` with the framework name.